hip-lang-config.cmake.in

```diff
@@ -0,0 +1,133 @@
+  get_filename_component(HIP_CLANG_INCLUDE_PATH "${HIP_CLANG_INCLUDE_PATH}" DIRECTORY)
+endif()
+
+#If HIP isnot installed under ROCm, need this to find HSA assuming HSA is under ROCm
+if( DEFINED ENV{ROCM_PATH} )
+  set(ROCM_PATH "$ENV{ROCM_PATH}")
+endif()
+
+#if HSA is not under ROCm then provide CMAKE_PREFIX_PATH=<HSA_PATH>
+find_path(HSA_HEADER hsa/hsa.h
+  PATHS
+    "${_IMPORT_PREFIX}/../include"
+    /opt/rocm/include
+)
+
+if (HSA_HEADER-NOTFOUND)
+  message (FATAL_ERROR "HSA header not found! ROCM_PATH environment not set")
+endif()
+
+set_target_properties(hip::device PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/../include;${HIP_CLANG_INCLUDE_PATH}>"
+  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/../include;${HIP_CLANG_INCLUDE_PATH}>"
+)
+
+if(HIP_RUNTIME MATCHES "ROCclr")
+  set_target_properties(hip::amdhip64 PROPERTIES
+    INTERFACE_COMPILE_DEFINITIONS "$<$<COMPILE_LANGUAGE:HIP>:__HIP_ROCclr__=1>"
+    INTERFACE_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/include;${HSA_HEADER}>"
+    INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/include;${HSA_HEADER}>"
+  )
+  set_target_properties(hip::device PROPERTIES
+    INTERFACE_COMPILE_DEFINITIONS "$<$<COMPILE_LANGUAGE:HIP>:__HIP_ROCclr__=1>"
+  )
+else()
+  set_target_properties(hip::hip_hcc_static PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/include;${HSA_HEADER}>"
+    INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/include;${HSA_HEADER}>"
+    )
+
+  set_target_properties(hip::hip_hcc PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/include;${HSA_HEADER}>"
+    INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "$<$<COMPILE_LANGUAGE:HIP>:${_IMPORT_PREFIX}/include;${HSA_HEADER}>"
+  )
+endif()
+
+if(HIP_COMPILER STREQUAL "clang")
+  set_property(TARGET hip::device APPEND PROPERTY
+    INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:HIP>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false>"
+  )
+
+  if (NOT EXISTS "${AMD_DEVICE_LIBS_PREFIX}/amdgcn/bitcode")
+    set_property(TARGET hip::device APPEND PROPERTY
+      INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:HIP>:--hip-device-lib-path=${AMD_DEVICE_LIBS_PREFIX}/lib>"
+    )
+  endif()
+
+  set_property(TARGET hip::device APPEND PROPERTY
+    INTERFACE_LINK_OPTIONS "$<$<LINK_LANGUAGE:HIP>:--hip-link>"
+  )
+
+  # Add support for __fp16 and _Float16, explicitly link with compiler-rt
+  set_property(TARGET hip::device APPEND PROPERTY
+    INTERFACE_LINK_LIBRARIES "$<$<LINK_LANGUAGE:HIP>:${HIP_CLANG_INCLUDE_PATH}/lib/linux/libclang_rt.builtins-x86_64.a>"
+  )
+endif()
+
+# Approved by CMake to use this name. This is used so that HIP can
+# change the name of the target and not require any modifications in CMake
+set(_CMAKE_HIP_DEVICE_RUNTIME_TARGET "hip::device")
```

Comment:
What does this exactly do? CMake will rename the target, is that so users can use `hip::device` from `find_package(hip)` as well? Dont we need hip-config.cmake to restrict the flags to CXX?

Reply:
> What does this exactly do?

The design point is to allow a level of flexibility between the two projects. By reading a a variable it allows the HIP package to elvove over time ( do a refactor ), and not break CMake backwards compat

>   is that so users can use `hip::device` from `find_package(hip)` as well.

See my comments below on de-duping the config files.

> Dont we need hip-config.cmake to restrict the flags to CXX?

Yes you will, but I don't know what HIP backwards compatibility rules are, and didn't want to presume that I could break existing users that rely on the current flag injection to make `C` files act as `HIP`.